### PR TITLE
Libhtp rs v18

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/issues/2696) ticket:
https://redmine.openinfosecfoundation.org/issues/2696

Describe changes:
- Use libhtp-rs.

Rebase from #12050 + CI checks `cargo fmt` for htp subdirectory

Draft to see CI/QA feedback with latest libhtp-rs changes, and not have too many conflicts

@jasonish how does the Cargo.toml logic look ?
Should we rather have a Cargo.toml.in ?

TODOs :
- look into more QA results
- look into the remaining oss-fuzz differential reports for libhtp-rs

### CODE REVIEW

**Comment by catenacyber** [comment]
@jasonish it looks like your repo https://github.com/jasonish/suricata-rpms needs an update for rust libhtp 
https://github.com/jasonish/suricata-rpms/blob/main/devel/suricata.spec#L190

**Comment by catenacyber** [comment]
No idea 
- why `DetectEngineHttpServerBodyFileDataTest21` fails now in some Windows (but not in previous PR nor in other systems)
- why macOS build now fails not finding lz4 (ok in previous PR)

**Comment by jasonish** [comment]
> @jasonish it looks like your repo https://github.com/jasonish/suricata-rpms needs an update for rust libhtp https://github.com/jasonish/suricata-rpms/blob/main/devel/suricata.spec#L190

Yes, noted. But I can only do that after merged here. The conditional logic allowed in an RPM spec file isn't flexible enough.

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -3096,7 +2981,7 @@ jobs:
       - run: CPATH="$HOMEBREW_PREFIX/include:$CPATH" LIBRARY_PATH="$HOMEBREW_PREFIX/lib:$LIBRARY_PATH" PATH="/opt/homebrew/opt/libtool/libexec/gnubin:$PATH" CFLAGS="${DEFAULT_CFLAGS}" ./configure  --enable-warnings --enable-unittests --prefix="$HOME/.local/"
       - run: CPATH="$HOMEBREW_PREFIX/include:$CPATH" LIBRARY_PATH="$HOMEBREW_PREFIX/lib:$LIBRARY_PATH" PATH="/opt/homebrew/opt/libtool/libexec/gnubin:$PATH" CFLAGS="${DEFAULT_CFLAGS}" make -j2
       # somehow it gets included by some C++ stdlib header (case unsensitive)
-      - run: rm libhtp/VERSION && make check
+      - run: make check
```
Comment (jasonish) [inline .github/workflows/builds.yml:2984]: For the MacOS build, as CPATH, LIBRARY_PATH aren't memorized by autoconf/automake you should provide them to every invocation of make.

Comment (catenacyber) [reply .github/workflows/builds.yml:2984]: Thanks

**Comment by catenacyber** [comment]
Converted QA pcaps to SV tests thanks to createst and FAILED:  427 to go

**Comment by suricata-qa** [comment]
WARNING:
 

ERROR: QA failed on SURI_TLPW1_files_sha256.

ERROR: QA failed on SURI_TLPR1_alerts_cmp.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_single_stats_chk | | |
| .uptime | 749 | 779 | 104.01% |
| | SURI_TLPW1_stats_chk | | |
| .app_layer.error.http.parser | 47 | 30 | 63.83% |
| | SURI_TLPR1_stats_chk | | |
| .uptime | 624 | 651 | 104.33% |
| .app_layer.tx.http | 5585895 | 5765118 | 103.21% |
| .app_layer.error.http.parser | 700 | 584 | 83.43% | 

Pipeline 23641

**Comment by catenacyber** [comment]
Next in https://github.com/OISF/suricata/pull/12214

**Pull request closed without merging**